src/services/parser/message/update-extract-for-sending.js

```diff
@@ -13,38 +8,37 @@ export const updateExtractForSending = async (
   sendingOdsCode
 ) => {
   const config = initializeConfig();
-  const parsedEhr = await new xml2js.Parser({ explicitArray: false }).parseStringPromise(
-    ehrExtract
-  );
+  const parsedEhr = await xmlStringToJsObject(ehrExtract);
 
   const sendingAsid = config.deductionsAsid;
 
   const rcmrUK06 = parsedEhr.RCMR_IN030000UK06;
   const controlActEvent = rcmrUK06.ControlActEvent;
 
-  rcmrUK06.communicationFunctionRcv.device = updateId(
-    rcmrUK06.communicationFunctionRcv.device,
-    receivingAsid
-  );
-  rcmrUK06.communicationFunctionSnd.device = updateId(
-    rcmrUK06.communicationFunctionSnd.device,
-    sendingAsid
-  );
-
-  controlActEvent.author1.AgentSystemSDS.agentSystemSDS = updateId(
-    controlActEvent.author1.AgentSystemSDS.agentSystemSDS,
-    sendingAsid
-  );
-  controlActEvent.subject.EhrExtract.inFulfillmentOf.priorEhrRequest.id['$'].root = ehrRequestId;
+  updateIdExtension(rcmrUK06.communicationFunctionRcv.device, receivingAsid);
+  updateIdExtension(rcmrUK06.communicationFunctionSnd.device, sendingAsid);
+  updateIdExtension(controlActEvent.author1.AgentSystemSDS.agentSystemSDS, sendingAsid);
+  controlActEvent.subject.EhrExtract.inFulfillmentOf.priorEhrRequest.id["@_root"] = ehrRequestId;
 
   updateAuthorOdsCode(controlActEvent.subject.EhrExtract, sendingOdsCode);
   updateAuthorOdsCode(controlActEvent.subject.EhrExtract.component.ehrFolder, sendingOdsCode);
 
-  const builder = new Builder();
-  return builder.buildObject(parsedEhr);
+  return jsObjectToXmlString(parsedEhr)
```

Comment:
Can we add a semicolon `;` here please?

```suggestion
  return jsObjectToXmlString(parsedEhr);
```